### ISSUE TITLE
Emacs ledger-mode hardcodes `--date-format`

### ISSUE THREAD

**Issue by codingquark** [simonmichael/hledger#1721]
In a [recent commit](https://github.com/ledger/ledger-mode/commit/85031ef8a6dc69aae78cb3242a92c001a4c12c07) ledger-mode started including `--date-format` every time it [checks for version](https://github.com/ledger/ledger-mode/blob/master/ledger-exec.el#L116).

When `ledger-binary-path` is set to hledger, opening a ledger file throws the following error:

```
hledger: Unknown flag: --date-format (use -h to see usage)
```

This is fairly reasonable since the `--date-format` option is ledgerism.

To solve this, preventing ledger from checking the version, fixes the issue:

```elisp
(setq ledger-mode-should-check-version nil)
```

Should this be reflected in the docs? Can we do something else about it?

**Comment by simonmichael** [comment]
Yes! Best of all would be a PR to ledger-mode, or failing that, gentle bug reports to at least discourage new Ledgerisms.

**Comment by acarrico** [comment]
This is an issue for more than just check-version, for example ```C-c C-l  ledger-display-ledger-stats```.

I guess a proxy script is necessary to fix this issue unless hledger decides to support ```--date-format``` (at least with the value ```%Y-%m-%d```):

ledger.sh:

```
#!/bin/sh

args=("$@")
for((i=0; i<"${#args[@]}"; ++i)); do
    case ${args[i]} in
        --date-format) unset args[i]; unset args[i+1]; break;;
    esac
done

hledger "${args[@]}"
```

Thanks to an Internet search for some option filtering shell code. Aren't shell script syntax and semantics insane? Of course, when you set up ledger-mode, use ```setq ledger-binary-path ledger.sh```  to install your script.

**Comment by simonmichael** [comment]
Thanks. I have inquired about the future at https://github.com/ledger/ledger-mode/issues/319

**Comment by simonmichael** [comment]
I'll close this as not directly a hledger issue. But please do contribute bug reports and fixes to ledger-mode to improve its hledger support (see https://github.com/ledger/ledger-mode/issues/319). It's a vital tool for some of us.
